### PR TITLE
implement periodic reprocessing of addons layers

### PR DESCRIPTION
**Pull request by paulcarlton-ww**

### CODE REVIEW

**Review by sbernheim — APPROVED** [review]
LGTM

File: pkg/main/main.go [hunk 1]
```diff
@@ -79,11 +81,23 @@ func main() {
 		"The address the health endpoint binds to.",
 	)
 
+	flag.StringVar(
```
Comment (richardcase) [inline pkg/main/main.go:84]: This should be `flag.DurationVar` instead

File: pkg/main/main.go [hunk 2]
```diff
@@ -79,11 +81,23 @@ func main() {
...
+		&syncPeriodStr,
+		"sync-period",
+		"30s",
```
Comment (richardcase) [inline pkg/main/main.go:87]: This can then be changed to `30*second` or something similar

File: pkg/main/main.go [hunk 3]
```diff
@@ -79,11 +81,23 @@ func main() {
...
+		"period between reprocessing of all AddonsLayers.",
+	)
+
 	flag.Parse()
 
+	syncPeriod, err := time.ParseDuration(syncPeriodStr)
```
Comment (richardcase) [inline pkg/main/main.go:93]: This can then be removed :)

**Comment by paulcarlton-ww** [comment]
I'll make this change

**Pull request merged**